config/config_test.go

```diff
@@ -98,7 +109,113 @@ func TestParseBoilerplateConfigOneVariableMissingName(t *testing.T) {
 	_, err := ParseBoilerplateConfig([]byte(CONFIG_ONE_VARIABLE_MISSING_NAME))
 
 	assert.NotNil(t, err)
-	assert.True(t, errors.IsError(err, VariableMissingName), "Expected a VariableMissingName error but got %s", reflect.TypeOf(err))
+	assert.True(t, errors.IsError(err, variables.RequiredFieldMissing("name")), "Expected a RequiredFieldMissing error but got %s: %v", reflect.TypeOf(errors.Unwrap(err)), err)
+}
+
+// YAML is whitespace sensitive, so we need to be careful that we don't introduce unnecessary indentation
+const CONFIG_ONE_VARIABLE_INVALID_TYPE =
+`variables:
+  - name: foo
+    type: foo
+`
+
+func TestParseBoilerplateConfigOneVariableInvalidType(t *testing.T) {
+	t.Parallel()
+
+	_, err := ParseBoilerplateConfig([]byte(CONFIG_ONE_VARIABLE_INVALID_TYPE))
+
+	assert.NotNil(t, err)
+	assert.True(t, errors.IsError(err, variables.InvalidBoilerplateType("foo")), "Expected a InvalidBoilerplateType error but got %s", reflect.TypeOf(errors.Unwrap(err)))
+}
+
+// YAML is whitespace sensitive, so we need to be careful that we don't introduce unnecessary indentation
+const CONFIG_ONE_VARIABLE_INVALID_TYPE_FOR_NAME_FIELD =
+`variables:
+  - name:
+      - foo
+      - bar
+`
+
+func TestParseBoilerplateConfigInvalidTypeForNameField(t *testing.T) {
+	t.Parallel()
+
+	_, err := ParseBoilerplateConfig([]byte(CONFIG_ONE_VARIABLE_INVALID_TYPE_FOR_NAME_FIELD))
+
+	assert.NotNil(t, err)
+	assert.True(t, errors.IsError(err, variables.InvalidTypeForField{FieldName: "name", ExpectedType: "string", ActualType: reflect.TypeOf([]interface{}{})}), "Expected a InvalidTypeForField error but got %s: %v", reflect.TypeOf(errors.Unwrap(err)), err)
+}
+
+// YAML is whitespace sensitive, so we need to be careful that we don't introduce unnecessary indentation
+const CONFIG_ONE_VARIABLE_ENUM_NO_OPTIONS =
+`variables:
+  - name: foo
+    type: enum
+`
+
+func TestParseBoilerplateConfigOneVariableEnumNoOptions(t *testing.T) {
+	t.Parallel()
+
+	_, err := ParseBoilerplateConfig([]byte(CONFIG_ONE_VARIABLE_ENUM_NO_OPTIONS))
+
+	assert.NotNil(t, err)
+	assert.True(t, errors.IsError(err, variables.OptionsMissing("foo")), "Expected a VariableMissingOptions error but got %s", reflect.TypeOf(errors.Unwrap(err)))
+}
+
+// YAML is whitespace sensitive, so we need to be careful that we don't introduce unnecessary indentation
+const CONFIG_ONE_VARIABLE_ENUM_OPTIONS_WRONG_TYPE =
+`variables:
+  - name: foo
+    type: enum
+    options: foo
+`
+
+func TestParseBoilerplateConfigOneVariableEnumWrongType(t *testing.T) {
+	t.Parallel()
+
+	_, err := ParseBoilerplateConfig([]byte(CONFIG_ONE_VARIABLE_ENUM_OPTIONS_WRONG_TYPE))
+
+	assert.NotNil(t, err)
+	assert.True(t, errors.IsError(err, variables.InvalidTypeForField{FieldName: "options", ExpectedType: "List", ActualType: reflect.TypeOf("string"), Context: "foo"}), "Expected a InvalidTypeForField error but got %s", reflect.TypeOf(errors.Unwrap(err)))
```

Comment:
The `options` field must be a list. If you wanted a list of one value, you would do:

```yaml
options:
  - foo
```